servers/physics_2d/godot_space_2d.cpp

```diff
@@ -686,6 +692,12 @@ bool GodotSpace2D::test_body_motion(GodotBody2D *p_body, const PhysicsServer2D::
 				break;
 			}
 
+			real_t inv_total_weight = 0.0;
+			for (int i = 0; i < cbk.amount; i++) {
+				inv_total_weight += wt[i];
+			}
+			inv_total_weight = Math::is_zero_approx(inv_total_weight) ? 1.0 : 1.0 / inv_total_weight;
```

Comment:
Rather than applying a recovery proportional to the weight, couldn't we exclude, for example at the second iteration, the collision shape with the lowest priority (to avoid the body, as seen in the first video, sinking a bit into the ground)?

Reply:
@reduz Do you have an opinion on this? It's more elegant the current solution, but if we exclude the recover for lower priority shape, it would avoid in the first video to entering the ground for a few frames and therefore, potentially recovering the information that the object is stuck in the ground (and also simply the behaviour -> recovers or not), I'm not sure.